doc/contribute/devtheory.rst

```diff
@@ -0,0 +1,185 @@
+Sense Tags
+----------
+
+Each surface is tagged with the two volume handles of the adjacent volumes.
+The first of the two surfaces is designated as the forward direction and the
+second is designated with the reverse direction. It is important to note that
+these surfaces senses may not be consistent with how an MC code determines
+the surface sense.
+
+Indices
+~~~~~~~
+
+Types of Indices
+----------------
+
+There are three ways to identify any surface and volume in a geometry:
+global ID, index, or entity handle. The global ID is an integer number that
+corresponds to the Cubit ID. As a result, the set of global IDs does not
+necessarily have to be contiguous. During setup, the global ID is tagged on the
+entity set. The second method is to identify by index. This is an ordinal
+numerical reference for each volume or surface in the geometry. The third method
+is to identify by the entity handle of the entity set. The entity handle is
+assigned by MOAB and is not necessarily contiguous.
+
+Cross Referencing
+-----------------
+
+If one type of indice is known, then either of the other two can be determined.
+By using the functions below or a combination of the functions, either ID,
+index, or entity handle can be determined. It is important to note that surfaces
+and volumes each have their own set of integer indexes and global IDs, while
+entity handles are unique to each entity set. Each integer (whether it is an
+index or global ID) can refer to multiple entity handles (a surface or volume).
+Therefore, when cross referencing by either index or global ID, the dimension
+needs to be supplied, but is not necessary when cross referencing by entity
+handle. There are five methods for cross referencing: entity handle by index
+(entity_by_index), global ID by index (id_by_index), entity handle by global
+ID (entity_by_id), global ID by entity handle (get_entity_id), index by entity
+handle (index_by_handle).
+
+OBB Tree
+~~~~~~~~
+
+What is an OBB?
+---------------
+
+An OBB, or Oriented Bounding Box, is a box that contains a single facet or set
+of facets and has an axis oriented such that it is the smallest possible box
+encompassing the facet(s). This is different from an axis aligned bounding box,
+which is aligned with the global axis and is not necessarily the smallest box
+possible. An example is the facet outlined in red below and it’s corresponding
+OBB. For the the rest of this section, facets will be represented by a line with
+a 2D OBB.
+
+..  image:: 3d-obb.png
+    :height: 300
+    :width:  300
+    :alt:    Image of a 3-D Oriented Bounding Box (OBB) around a facet
+
+OBB Tree Construction
+---------------------
+
+First, the OBB tree for each surface is built. This is built top down where the
+top level is an OBB that contains all the surface’s facets. Then the set of facets
+is split roughly in half and two new OBBs form around each of the new sets.
+This continues until each of the OBBs the level contains a single facet. A 2D
```

Comment:
Not necessary to alter the existing image though. I'd just be vague and indicate that it contains a small number of facets.